### DISCUSSION TITLE
Adding fields to <model>CreateInput InputType Class

### DISCUSSION THREAD

**Discussion by MatheusGNachtigall** [MichalLytek/typegraphql-prisma#409] [Q&A] [answered]
Hello.

Apologies if this is a dumb question or if it has been answered somewhere already...I'm just starting with graphql and everything is very new to me.

Is it possible to have extra fields in the createInput class that are not on the prisma schema?

Lets say I have a model like this:

```
model User {
  email   String
  id      String   @id @default(uuid())
}
```

after running prisma generate I do get the following UserCreateInput 

```
@TypeGraphQL.InputType('UserCreateInput', {})
export class UserCreateInput {

	@TypeGraphQL.Field((_type) => String, {
		nullable: false,
	})
	email!: string;

	@TypeGraphQL.Field((_type) => String, {
		nullable: true,
	})
	id?: string | undefined;
}
```

Lets say I want to also send the userPassword (and register it somewhere else...like passport/identity/etc) via graphql

I know I can get that by manually adding

```
	@TypeGraphQL.Field((_type) => String, {
		nullable: true,
	})
	password?: string | undefined;
```
to the createInput class...but that would be overwritten next time I ran prisma generate.

I checked this link but it only seems to work for the "User" resolver... not "UserCreateInput"
https://prisma.typegraphql.com/docs/advanced/adding-fields/

Am I missing something?

Thanks in advance

**Comment by MichalLytek** [comment, marked as answer]
https://typegraphql.com/docs/inheritance.html#types-inheritance

**Reply by MatheusGNachtigall** [reply]
Thank you so much!

Was able to get it working properly!

Great job at developing these tools!
